### PR TITLE
[CDAP-7493] Use a backward-compatibe deserializer for DatasetModuleId…

### PR DESCRIPTION
**Pull request by anew**
… in DatasetTypeMDS. This will avoid requiring an upgrade step for the DatasetTypeMDS.

### CODE REVIEW

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by anew** [comment]
Waiting for build before I merge: http://builds.cask.co/browse/CDAP-RUT349

**Comment by anew** [comment]
Build passed. Squashing and merging.

**Pull request merged**
